source/game/system/map/MapdataCheckPoint.cc

```diff
@@ -115,8 +220,7 @@ void MapdataCheckPointAccessor::init() {
         s8 checkArea = checkpoint->checkArea();
         checkpoint->initCheckpointLinks(*this, ckptId);
 
-        if (static_cast<MapdataCheckPoint::CheckArea>(checkArea) ==
-                MapdataCheckPoint::CheckArea::FINISH_LINE) {
+        if (checkpoint->isFinishLine()) {
```

Comment:
```suggestion
        if (checkpoint->isFinishLine()) {
```
This is dependent on if the checkArea is ever set in `initCheckpointLinks`. If it is, this helper function isn't useful. If it's not, the above line assigning checkArea can be removed.

Reply:
`initCheckpointLinks` does not modify `m_checkArea` so we can remove the above line